From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jolla.com>
Date: Thu, 6 Aug 2026 12:07:11 +0200
Subject: [PATCH 06/67] Fix GLContextProvider defines

Avoid declaring the configured EGL provider twice now that ESR140 also
exposes EGL unconditionally.

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
Signed-off-by: Pavel Tumakaev <p.tumakaev@omprussia.ru>
---
 gfx/gl/GLContextProvider.h | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/gfx/gl/GLContextProvider.h b/gfx/gl/GLContextProvider.h
index 277b76efa856e..38cd5e2135a53 100644
--- a/gfx/gl/GLContextProvider.h
+++ b/gfx/gl/GLContextProvider.h
@@ -44,9 +44,11 @@ namespace gl {
 #  define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderCGL
 #endif
 
-#define GL_CONTEXT_PROVIDER_NAME GLContextProviderEGL
-#include "GLContextProviderImpl.h"
-#undef GL_CONTEXT_PROVIDER_NAME
+#ifndef MOZ_GL_PROVIDER
+#  define GL_CONTEXT_PROVIDER_NAME GLContextProviderEGL
+#  include "GLContextProviderImpl.h"
+#  undef GL_CONTEXT_PROVIDER_NAME
+#endif
 
 #if defined(MOZ_WIDGET_GTK)
 #  ifdef MOZ_X11
@@ -69,14 +71,17 @@ namespace gl {
 #  endif
 #endif
 
-#ifndef GL_CONTEXT_PROVIDER_DEFAULT
-#  define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderEGL
+#ifndef MOZ_GL_PROVIDER
+#  ifndef GL_CONTEXT_PROVIDER_DEFAULT
+#    define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderEGL
+#  endif
 #endif
 
 #ifdef MOZ_GL_PROVIDER
 #  define GL_CONTEXT_PROVIDER_NAME MOZ_GL_PROVIDER
 #  include "GLContextProviderImpl.h"
 #  undef GL_CONTEXT_PROVIDER_NAME
+#  undef GL_CONTEXT_PROVIDER_DEFAULT
 #  define GL_CONTEXT_PROVIDER_DEFAULT MOZ_GL_PROVIDER
 #endif
 
